src/Init/Data/List/Zip.lean

```diff
@@ -418,6 +418,36 @@ theorem unzip_eq_map : ∀ {l : List (α × β)}, unzip l = (l.map Prod.fst, l.m
   | [] => rfl
   | (a, b) :: l => by simp only [unzip_cons, map_cons, unzip_eq_map (l := l)]
 
+theorem getElem?_fst_unzip {l : List (α × β)} {i : Nat} :
+    (unzip l).fst[i]? = l[i]?.map Prod.fst := by simp
+
+theorem getElem?_snd_unzip {l : List (α × β)} {i : Nat} :
+    (unzip l).snd[i]? = l[i]?.map Prod.snd := by simp
+
+theorem getElem?_fst_unzip_eq_some {l : List (α × β)} {i : Nat} :
+    (unzip l).fst[i]? = some a ↔ ∃ b, l[i]? = some (a, b) := by simp
+
+theorem getElem?_snd_unzip_eq_some {l : List (α × β)} {i : Nat} :
+    (unzip l).snd[i]? = some b ↔ ∃ a, l[i]? = some (a, b) := by simp
+
+theorem length_fst_unzip {l : List (α × β)} :
+    length (unzip l).fst = l.length := by simp
+
+theorem length_snd_unzip {l : List (α × β)} :
+    length (unzip l).snd = l.length := by simp
```

Comment:
I think the point of the `unzip_fst` lemma above is that it removes the need for any further lemmas about `(unzip l).fst`, and so the omission of all of these (and perhaps the rest of the file too) is deliberate.

Reply:
I was told by Kim that she did want them and to make a PR.

Reply:
https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Array.20sorting.2E